common/djangoapps/third_party_auth/tests/test_pipeline_integration.py

```diff
@@ -336,3 +334,52 @@ def test_provider_settings_redirect_to_registration(self, send_to_registration_f
                 )
                 assert response.status_code == 302
                 assert response.url == expected_redirect_url
+
+
+@unittest.skipUnless(testutil.AUTH_FEATURE_ENABLED, testutil.AUTH_FEATURES_KEY + ' not enabled')
+class UserDetailsForceSyncTestCase(testutil.TestCase, test.TestCase):
+    """Tests to ensure learner profile data is properly synced if the provider requires it."""
+
+    def setUp(self):
+        super(UserDetailsForceSyncTestCase, self).setUp()
+        self.user = UserFactory.create()  # pylint: disable=attribute-defined-outside-init
+        self.details = {
+            'email': 'new+{}'.format(self.user.email),
+            'username': 'new_{}'.format(self.user.username),
+            'fullname': 'Grown Up {}'.format(self.user.profile.name),
+            'country': 'PK',
+            'non_existing_field': 'value',
+        }
+
+        # Mocks
+        self.strategy = mock.MagicMock()
+
+        get_from_pipeline = mock.patch('third_party_auth.pipeline.provider.Registry.get_from_pipeline')
+        self.get_from_pipeline = get_from_pipeline.start()
+        self.get_from_pipeline.return_value = mock.MagicMock(sync_learner_profile_data=True)
```

Comment:
@UmanShahzad I'm assuming that another test (implicitly) covers the case where `sync_learner_profile_data` is `False`? Would it be worth adding a test that explicitly checks that data doesn't get updated in that case?

Reply:
Yeah, close to a hundred tests probably cover that case. It wouldn't be worth checking it explicitly here.